zerver/lib/bugdown/api_arguments_table_generator.py

```diff
@@ -39,19 +43,31 @@ def run(self, lines: List[str]) -> List[str]:
                 match = REGEXP.search(line)
 
                 if match:
-                    json_filename = match.group(1)
-                    doc_filename = match.group(2)
-                    json_filename = os.path.expanduser(json_filename)
-                    if not os.path.isabs(json_filename):
-                        json_filename = os.path.normpath(os.path.join(self.base_path, json_filename))
+                    filename = match.group(1)
+                    doc_name = match.group(2)
+                    filename = os.path.expanduser(filename)
+                    if not os.path.isabs(filename):
+                        if filename.endswith('.json'):
+                            parent_dir = self.base_path
+                        elif filename.endswith('.yaml'):
+                            parent_dir = self.openapi_path
+                        filename = os.path.normpath(os.path.join(parent_dir, filename))
+
                     try:
-                        with open(json_filename, 'r') as fp:
-                            json_obj = ujson.load(fp)
-                            arguments = json_obj[doc_filename]
+                        with open(filename, 'r') as fp:
+                            if filename.endswith('.json'):
+                                json_obj = ujson.load(fp)
+                                arguments = json_obj[doc_name]
+                            elif filename.endswith('.yaml'):
+                                yaml_parser = YamoleParser(fp)
+                                yaml_obj = yaml_parser.data
+                                path, method = doc_name.rsplit(':', 1)
+                                arguments = yaml_obj['paths'][path][method]['parameters']
```

Comment:
I don't like the feel of this block, where you repeat the check on `filename` to figure out what parser to use.  I'd much prefer saving the answer near the top of this block as `is_openapi_format = filename.endswith('.yaml')` and then checking that here and in the other conditional.

Reply:
Yeah, I've had this doubt as well. I did it this way so it's easier to remove whenever we end up switching to the final format, but you're right: that looks better.